pkg/argparse/args.go

```diff
@@ -0,0 +1,288 @@
+/*
+ * Copyright 2024 CloudWeGo Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package argparse
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/kitex-contrib/kitexcall/pkg/config"
+	"github.com/kitex-contrib/kitexcall/pkg/errors"
+	"github.com/kitex-contrib/kitexcall/pkg/log"
+)
+
+type Argument struct {
+	config.Config
+	help bool
+}
+
+func NewArgument() *Argument {
+	return &Argument{
+		Config: config.Config{
+			Meta:           make(map[string]string),
+			MetaPersistent: make(map[string]string),
+		},
+	}
+}
+
+type EndpointList []string
+
+func (e *EndpointList) String() string {
+	return strings.Join(*e, ",")
+}
+
+func (e *EndpointList) Set(value string) error {
+	*e = append(*e, value)
+	return nil
+}
+
+type KVMap map[string]string
+
+func (m *KVMap) String() string {
+	return ""
+}
+
+func (m *KVMap) Set(value string) error {
+	parts := strings.SplitN(value, "=", 2)
+	if len(parts) != 2 {
+		return errors.New(errors.ArgParseError, "Invalid format for key-value pair: %s", value)
+	}
+	(*m)[parts[0]] = parts[1]
+	return nil
+}
+
+func (a *Argument) buildFlags() *flag.FlagSet {
+	f := flag.NewFlagSet(os.Args[0], flag.ContinueOnError)
+
+	f.BoolVar(&a.help, "help", false, `Print usage instructions and exit.`)
+	f.BoolVar(&a.help, "h", false, `Print usage instructions and exit.`)
+
+	f.StringVar(&a.Type, "type", "unknown", "Specify the type of IDL: 'thrift' or 'protobuf'.")
+	f.StringVar(&a.Type, "t", "unknown", "Specify the type of IDL: 'thrift' or 'protobuf'. (shorthand)")
+
+	f.StringVar(&a.IDLPath, "idl-path", "", "Specify the path of IDL file.")
+	f.StringVar(&a.IDLPath, "p", "", "Specify the path of IDL file. (shorthand)")
+
+	f.StringVar(&a.Method, "method", "", "Specify the method name in the format `ServiceName/MethodName` or or just `MethodName`, such as `GenericService/ExampleMethod`")
```

Comment:
"IDLServiceName" makes more sense than "ServiceName" here (and the next line).

Reply:
done